var/spack/repos/builtin/packages/py-scikit-learn/package.py

```diff
@@ -51,43 +52,38 @@ class PyScikitLearn(PythonPackage):
     variant("openmp", default=True, description="Build with OpenMP support")
 
     # setup.py
-    depends_on("python@2.6:2.8,3.3:", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.4:", when="@0.20:", type=("build", "run"))
-    depends_on("python@3.5:", when="@0.21:", type=("build", "run"))
-    depends_on("python@3.6:", when="@0.23:", type=("build", "run"))
-    depends_on("python@3.7:", when="@1.0:", type=("build", "run"))
     depends_on("python@3.8:", when="@1.1:", type=("build", "run"))
```

Comment:
Should this be a general requirement (to reflect earlier versions of the package that are still included above)?

Reply:
I'm on the fence. Spack no longer supports Python 3.6 and older, so as far as I'm concerned, all older versions of py-scikit-learn support all older versions of Python (that Spack supports). That's why I removed the historical record. In the past, we kept this in in case someone used an external version of a dependency that was older. I'm not sure whether I want to be short and simple or long and complete here.

Reply:
FWIW. I'm on a machine at LLNL whose default is still `2.7.16`.  I know Spack  doesn't support older so just providing a data point.

Reply:
Oh yeah, it isn't hard to find Python 2, but if someone tries to use it and finds that it doesn't work, I'm no longer supporting it.